src/uwtools/config/validator.py

```diff
@@ -45,6 +42,23 @@ def validate_yaml(schema_file: DefinitePath, config_file: OptionalPath = None) -
 # Private functions
 
 
+def _prep_config(config: Union[dict, YAMLConfig, OptionalPath]) -> YAMLConfig:
+    """
+    Ensure a dereferenced YAMLConfig object for various input types.
+
+    :param config: The config to validate.
+    :return: A dereferenced YAMLConfig object based on the input config.
+    """
+    if isinstance(config, dict):
+        cfgobj = YAMLConfig(config=config)
+    elif isinstance(config, YAMLConfig):
+        cfgobj = config
+    else:
+        cfgobj = YAMLConfig(config=config)
```

Comment:
Am I missing an edge case where this wouldn't reduce?

```
    if isinstance(config, YAMLConfig):
        cfgobj = config
    else:
        cfgobj = YAMLConfig(config=config)
```

Reply:
I think you're right!